File: src/2025/day08/src/test/java/ca/terrylockett/aoc2025/day08/TestDay08.java
package ca.terrylockett.aoc2025.day08;

import ca.terrylockett.aoccommon.input.PuzzleInput;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TestDay08 {

	static final String TEST_FILE_NAME = "test-input.txt";

	static String testInput = "";

	@BeforeAll
	static void setup() {
		testInput = PuzzleInput.getInput(TEST_FILE_NAME).orElseThrow();
	}

	@Test
	void testDay08part1() {
		assertEquals(40, Day08Runner.part1(testInput, 10));
	}

	@Test
	void testDay08part2() {
		assertEquals(25272, Day08Runner.part2(testInput));
	}
}

File: src/2023/day08/src/test/kotlin/ca/terrylockett/aoc2023/day08/TestDay08.kt
package ca.terrylockett.aoc2023.day08

import ca.terrylockett.aoccommon.resources.Resources
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TestDay08 {

	@Test
	fun part1_1() {
		val inputFilePath = Resources.getInputFilePath("test-input.txt").orElseThrow()
		assertEquals(2, part1(inputFilePath))
	}

	@Test
	fun part1_2() {
		val inputFilePath = Resources.getInputFilePath("test-input2.txt").orElseThrow()
		assertEquals(6, part1(inputFilePath))
	}

//    @Test
//    fun part2() {
//        val inputFilePath = Resources.getInputFilePath("test-input.txt").orElseThrow()
//        assertEquals(0, 0)
//    }
}
